web/src/user_events.js

```diff
@@ -128,6 +129,38 @@ export const update_person = function update(person) {
 
     if (Object.hasOwn(person, "custom_profile_field")) {
         people.set_custom_profile_field_data(person.user_id, person.custom_profile_field);
+        const $navbar_alert = $("#navbar_alerts_wrapper").children(".alert").first();
+        if (
+            !$navbar_alert?.length ||
+            $navbar_alert.is("#empty-required-profile-fields-warning") ||
+            $navbar_alert.is(":hidden")
+        ) {
+            navbar_alerts.maybe_show_empty_required_profile_fields_alert();
+        }
```

Comment:
This block is repeated in a few places; I think it should probably be a function exported by the `navbar_alerts` module. 

Or maybe just part of the `navbar_alerts.maybe_show_empty_required_profile_fields_alert` interface? 

In any case, you probably also only need to do this check if `person.user_id == current_user.user_id`, and that seems like an appropriate outer conditional.

Reply:
I have transferred that block in ```navbar_alerts.maybe_show_empty_required_profile_fields_alert``` now.